test/with_ar/all_agnostic_test.rb

```diff
@@ -438,6 +438,32 @@ def test_format
         end
       end
 
+      def test_format_iso_week
+        skip "Unsupported ISO week number for DB=#{ENV['DB']}" if ['sqlite'].include?(ENV['DB'])
+        assert_equal '10', t(@lucas, @updated_at.format('%V'))
+
+        assert_equal '01', t(@lucas, Arel::Nodes.build_quoted('2024-01-01 10:42:00').cast(:datetime).format('%V')) # Monday
+        assert_equal '01', t(@lucas, Arel::Nodes.build_quoted('2030-01-01 10:42:00').cast(:datetime).format('%V')) # Tuesday
+        assert_equal '01', t(@lucas, Arel::Nodes.build_quoted('2025-01-01 10:42:00').cast(:datetime).format('%V')) # Wednesday
+        assert_equal '01', t(@lucas, Arel::Nodes.build_quoted('2026-01-01 10:42:00').cast(:datetime).format('%V')) # Thursday
+        assert_equal '53', t(@lucas, Arel::Nodes.build_quoted('2027-01-01 10:42:00').cast(:datetime).format('%V')) # Friday
+        assert_equal '52', t(@lucas, Arel::Nodes.build_quoted('2028-01-01 10:42:00').cast(:datetime).format('%V')) # Saturday
+        assert_equal '52', t(@lucas, Arel::Nodes.build_quoted('2034-01-01 10:42:00').cast(:datetime).format('%V')) # Sunday
+      end
+
+      def test_format_iso_year_of_week
+        skip "Unsupported ISO year of week for DB=#{ENV['DB']}" if ['mssql', 'sqlite'].include?(ENV['DB'])
+        assert_equal '2014', t(@lucas, @updated_at.format('%G'))
+
+        assert_equal '2024', t(@lucas, Arel::Nodes.build_quoted('2024-01-01 10:42:00').cast(:datetime).format('%G')) # Monday
```

Comment:
```ruby
{
  '2024-01-01 10:42:00' => '2024', # Monday
  '2030-01-01 10:42:00' => '2030', # Tuesday
  '2025-01-01 10:42:00' => '2025', # Wednesday
  '2026-01-01 10:42:00' => '2026', # Thursday
  '2027-01-01 10:42:00' => '2026', # Friday
  '2028-01-01 10:42:00' => '2027', # Saturday
  '2034-01-01 10:42:00' => '2033', # Sunday
}.each do |date, exp|
  assert_equal exp, t(@lucas, Arel::Nodes.build_quoted(date).cast(:datetime).format('%V'))
end
```
Or introducing some ancillary functions to avoid the long repetitions.

Or possibly have a constant with the expected results:
```ruby
{
  '2024-01-01 10:42:00' => {'%G' => '2024', '%V' => '01',...}, # Monday
  '2030-01-01 10:42:00' => {'%G' => '2030', '%V' => '01',...}, # Tuesday
  '2025-01-01 10:42:00' => {'%G' => '2025', '%V' => '01',...}, # Wednesday
  '2026-01-01 10:42:00' => {'%G' => '2026', '%V' => '01',...}, # Thursday
  '2027-01-01 10:42:00' => {'%G' => '2026', '%V' => '53',...}, # Friday
  '2028-01-01 10:42:00' => {'%G' => '2027', '%V' => '52',...}, # Saturday
  '2034-01-01 10:42:00' => {'%G' => '2033', '%V' => '52',...}, # Sunday
}

Reply:
This does look better indeed, but isn't it better to separate them for better error messages?

Reply:
I don't see what difference it would make in the error messages.  The calls to assert are exactly the same.